sentinelsat/sentinel.py

```diff
@@ -755,7 +755,7 @@ def _md5_compare(self, file_path, checksum, block_size=2 ** 13):
                     progress.update(len(block_data))
             return md5.hexdigest().lower() == checksum.lower()
 
-    def _download(self, url, path, session, file_size):
+    def _download(self, url, path, session, file_size, speed_boundary_in_kb=1):
```

Comment:
* please change the name from "speed_boundary_in_kb" to "timeout"
* timeout should be a class parameter that is provided with the class, not a function parameter
* documentation is necessary in the class docstring that the timeout refers to seconds per 1MB chunk